bootstrap/src/sun/nio/ch/lincheck/Injections.java

```diff
@@ -641,6 +641,24 @@ public static void onInlineMethodCallException(int methodId, Throwable t) {
         getEventTracker().onInlineMethodCallException(methodId, t);
     }
 
+    /**
+     * Called at the beginning of every loop iteration (including the first one).
+     */
+    public static void onLoopIteration(int codeLocation, int loopId) {
+        getEventTracker().onLoopIteration(codeLocation, loopId);
+    }
+
+    /**
+     * Called on a normal (non-exceptional) exit from a loop body and
+     * at an exception handler entry that is reachable from within a loop body and lies outside it.
+     *
+     * @param isReachableFromOutsideLoop true if the handler can also be reached from outside the loop body;
+     *   false if it is exclusive to the loop body.
+     */
```

Comment:
Update the documentation (can be called on both normal and exceptional exits), and also document `exception` parameter.